### DISCUSSION TITLE
Failed OIDC callback due to missing nonce in Keycloak

### DISCUSSION THREAD

**Discussion by JeffreyHeneine** [keycloak/keycloak#19986] [Q&A] [answered]
I am currently facing an issue with the integration of ZOHO People API as Identity Provider (OIDC) in Keycloak. When attempting to make an identity provider OAuth callback, the following error message is displayed:

"Failed to make identity provider OAuth callback: org.keycloak.broker.provider.IdentityBrokerException: OpenID Provider [OIDC] did not return a nonce."

I have checked the Keycloak documentation and forums, but I have not found any solutions to this problem.

Could you please provide me with any insights or suggestions on how to resolve this issue? Additionally, if there are any further details that would be helpful in diagnosing the problem, please let me know.

Thank you for your assistance.

**Comment by jbman** [comment, marked as answer]
If a nonce URL parameter is provided to the authorization URL, the Identity Provider must return in a nonce claim with the provided value.
This is required for ID token validation, see https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation

Please check if ZOHO People API can provide the nonce claim in the ID token.
